runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -349,15 +349,32 @@ public void prefixTrim(final Token address) {
                                     .map(e::getLogUnitClient)
                                     .map(client -> client.prefixTrim(address))
                                     .forEach(CFUtils::getUninterruptibly);
-                            return null;    // No return value
-                        }, true);
+                            return null;
+                }, true);
                 // TODO(Maithem): trimCache should be epoch aware?
                 runtime.getSequencerView().trimCache(address.getSequence());
                 break;
             } catch (NetworkException | TimeoutException e) {
                 log.warn("prefixTrim: encountered a network error on try {}", x, e);
                 Duration retryRate = runtime.getParameters().getConnectionRetryRate();
                 Sleep.sleepUninterruptibly(retryRate);
+            } catch (RuntimeException re) {
```

Comment:
I believe you catch RuntimeException and not WrongEpochException because the server doesn't return the correct error message, but fails and the "uncaught exception handler" wraps the throwable in a RuntimeException and sends it back and then you need to unwrap it as such.

I think you should just fix the server side and just catch WrongEpochException

Reply:
I'm not clear with this. Even if we catch this on the server side, we can't do anything as it's the client who must move to the new epoch and retry. Are you proposing sending a new kind of message?